x/httpproxy/connect_handler.go

```diff
@@ -35,11 +38,12 @@ func (h *handler) ServeHTTP(proxyResp http.ResponseWriter, proxyReq *http.Reques
 	if proxyReq.Method == http.MethodConnect {
 		h.handleConnect(proxyResp, proxyReq)
 		return
-	} else if proxyReq.URL.Host != "" {
+	}
+	if proxyReq.URL.Host != "" {
 		h.handleHTTPProxyRequest(proxyResp, proxyReq)
-	} else {
-		http.Error(proxyResp, "Not Found", http.StatusNotFound)
+		return
 	}
+	http.Error(proxyResp, "Not Found", http.StatusNotFound)
```

Comment:
*nit* it's a little bit weird that the normal path resulted in an error, maybe "early return" the `Not Found` first.

Reply:
The problem is that we have two normal paths and one error, which is a fallback from the other two, so we can't do it first.